uPortal-soffit/uPortal-soffit-renderer/src/main/java/org/apereo/portal/soffit/security/SoffitApiPreAuthenticatedProcessingFilter.java

```diff
@@ -0,0 +1,112 @@
+ * Security) allows REST APIs within Soffits and similar modules to know the identity of a portal
+ * users simply and securely. The approach is inspired by (and closely aligned with) OpenID Connect
+ * (OIDC). This filter essentially requests and receives an OIDC ID Token from the portal. The token
+ * is available if the user has previously signed in.
+ *
+ * @since 5.1
+ */
+public class SoffitApiPreAuthenticatedProcessingFilter
+        extends AbstractPreAuthenticatedProcessingFilter {
+
+    private static final String USER_DETAILS_REQUEST_ATTRIBUTE =
+            SoffitApiPreAuthenticatedProcessingFilter.class.getName()
+                    + ".userDetailsRequestAttribute";
+
+    @Value(
+            "${"
+                    + AbstractJwtService.SIGNATURE_KEY_PROPERTY
+                    + ":"
+                    + AbstractJwtService.DEFAULT_SIGNATURE_KEY
+                    + "}")
+    private String signatureKey;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public SoffitApiPreAuthenticatedProcessingFilter() {
+        // Check we have the correct principal on each request
+        setCheckForPrincipalChanges(true);
+    }
+
+    @PostConstruct
+    public void init() {
+        // Log a warning if a custom signature key is not specified...
+        if (AbstractJwtService.DEFAULT_SIGNATURE_KEY.equals(signatureKey)) {
+            logger.warn(
+                    "A custom value for '{}' has not been specified;  the default value will be used.  This configuration is not production-safe!",
+                    AbstractJwtService.SIGNATURE_KEY_PROPERTY);
+        }
+    }
+
+    @Override
+    protected Object getPreAuthenticatedPrincipal(HttpServletRequest request) {
+
+        final String authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (StringUtils.isBlank(authHeader)
+                || !authHeader.startsWith(Headers.BEARER_TOKEN_PREFIX)) {
+            /*
+             * Nothing we can do if (1) the Authorization header is not set or (2) the value isn't a
+             * Bearer token.
+             */
+            return null;
+        }
+
+        final String bearerToken = authHeader.substring(Headers.BEARER_TOKEN_PREFIX.length());
+
+        try {
+            // Validate & parse the JWT
+            final Jws<Claims> claims =
+                    Jwts.parser().setSigningKey(signatureKey).parseClaimsJws(bearerToken);
+
+            logger.debug("Found the following pre-authenticated user:  {}", claims.toString());
+
+            final UserDetails rslt = new SoffitApiUserDetails(claims.getBody().getSubject());
+            request.setAttribute(USER_DETAILS_REQUEST_ATTRIBUTE, rslt);
+            return rslt;
+        } catch (Exception e) {
+            logger.error(
+                    "Unable to identify pre-authenticated user with Bearer token '{}'",
+                    bearerToken,
+                    e);
```

Comment:
A stack trace is probably too much to log at `ERROR` level for something as trivial as an expired token.